clar2wasm/src/wasm_utils.rs

```diff
@@ -1301,7 +1310,16 @@ pub fn call_function<'a>(
                 };
             }
 
-            error_mapping::from_runtime_error_code(instance, &mut store)
+            // Check if the error is caused by a Wasm trap.
+            //
+            // In this case, runtime errors are handled
+            // by being mapped to the corresponding ClarityWasm Errors.
+            if let Some(_trap) = e.root_cause().downcast_ref::<Trap>() {
+                return error_mapping::from_runtime_error_code(instance, &mut store);
+            }
```

Comment:
We should certainly not catch all the [Trap errors](https://docs.rs/wasmtime/latest/wasmtime/enum.Trap.html) indistinctly. For example, there is no reason an out of bond access to a table should be processed by `from_runtime_error_code`: it will just find the default error code `-1` and tells us it's not mapped, which would be meaningless.

Reply:
Changed to consider only `UnreachableCodeReached` trap.